src/dotty/tools/dotc/transform/LazyVals.scala

```diff
@@ -124,7 +121,12 @@ class LazyVals extends MiniPhaseTransform with SymTransformer {
 
 
   override def transformStats(trees: List[tpd.Tree])(implicit ctx: Context, info: TransformerInfo): List[tpd.Tree] = {
-    val (holders, stats) = trees.partition { _.symbol.flags == containerFlags}
+    val (holders, stats) =
+      atGroupEnd { implicit ctx: Context =>
+        trees.partition {
+          _.symbol.flags == containerFlags
```

Comment:
If they are re-ordered that depends crucially on the fact that nobody looked at the info, name, flags, or owner of the `holder` symbols yet. Because any read of these fields would cause the Touched flag to be set.